<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Spatial Experiment</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sean Pierre-Louis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="research.html">
    <span class="fa fa-laptop"></span>
     
    Research
  </a>
</li>
<li>
  <a href="CV.html">
    <span class="fa fa-dna"></span>
     
    CV
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="mailto:Sean.Pierre-Louis@unh.edu">
    <span class="fa fa-envelope-o fa"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/SeanPie">
    <span class="fa fa-github fa"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/seanpierre_">
    <span class="fa fa-twitter fa"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/sean-pierre-louis-0a5ab3a9/">
    <span class="fa fa-linkedin"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Spatial Experiment</h1>

</div>


<p><strong>Spatial Transcriptomics analysis based on <a
href="https://lmweber.org/OSTA-book/analysis-steps.html"
class="uri">https://lmweber.org/OSTA-book/analysis-steps.html</a></strong></p>
<p>For both practice data from github and from mice kidney paper.</p>
<div id="spatial-experiment-object-notes" class="section level1">
<h1><a href="SpatialVis.html">Spatial Experiment Object Notes</a></h1>
</div>
<div id="load-the-data" class="section level1">
<h1>Load the data</h1>
<p>Create a Spatial Experiment object.</p>
<pre class="r"><code>dir &lt;- file.path(&quot;~&quot;,&quot;Desktop&quot;,&quot;test&quot;)

sample1 &lt;-  file.path(dir, &quot;sham&quot;, &quot;outs&quot;)

shamKidney &lt;- read10xVisium(sample1,
                        type = &quot;sparse&quot;, data = &quot;filtered&quot;,
                        images = &quot;lowres&quot;, load = FALSE)</code></pre>
<p>Spatial Experiment objects can created with merged data from multiple
tissue samples.</p>
<pre class="r"><code>sample_ids2 &lt;- c(&quot;2day&quot;, &#39;12hour&#39;)
samples2 &lt;- file.path(dir, sample_ids2, &quot;outs&quot;)

samples_idsAll &lt;- c(&#39;2day&#39;, &#39;12hour&#39;, &#39;week&#39;, &#39;sham&#39;)
samplesAll &lt;-  file.path(dir, samples_idsAll, &#39;outs&#39;)

#Import visium data for mutiple samples
sham_6weekKidney &lt;- read10xVisium(samples2, sample_ids2,
                        type = &quot;sparse&quot;, data = &quot;filtered&quot;,
                        images = &quot;lowres&quot;, load = FALSE)

allKidneys &lt;- read10xVisium(samplesAll, samples_idsAll,
                        type = &quot;sparse&quot;, data = &quot;filtered&quot;,
                        images = &quot;lowres&quot;, load = FALSE)</code></pre>
</div>
<div id="pre-processing" class="section level1">
<h1>Pre-Processing</h1>
<pre class="r"><code>plotSpots(shamKidney)
plotSpots(sham_6weekKidney)
plotSpots(allKidneys)
plotVisium(allKidneys)
plotVisium(shamKidney)</code></pre>
<p><img src="SpatialVis_files/figure-html/figures-side-1.png" width="50%" /><img src="SpatialVis_files/figure-html/figures-side-2.png" width="50%" /><img src="SpatialVis_files/figure-html/figures-side-3.png" width="50%" /><img src="SpatialVis_files/figure-html/figures-side-4.png" width="50%" /><img src="SpatialVis_files/figure-html/figures-side-5.png" width="50%" /></p>
</div>
<div id="filtering-and-normalization" class="section level1">
<h1>Filtering and Normalization</h1>
<pre class="r"><code># add some values in &#39;colData&#39; to annotate spots, also done by addPerCellQC function, 
# colData(shamKidney)$sum &lt;- colSums(counts(shamKidney))

#All spots already in tissue because using filtered data
shamKidney &lt;- shamKidney[, colData(shamKidney)$in_tissue == TRUE]

#Get a list of gene names using rowData, symbol columns is the names
head(rowData(shamKidney)$symbol)</code></pre>
<pre><code>## [1] &quot;Xkr4&quot;    &quot;Gm1992&quot;  &quot;Gm19938&quot; &quot;Gm37381&quot; &quot;Rp1&quot;     &quot;Sox17&quot;</code></pre>
<pre class="r"><code>#Used to look for specific genes, in this example searches for mitochondrial genes
is_mito &lt;- grepl(&quot;(^MT-)|(^mt-)&quot;, rowData(shamKidney)$symbol) 

table(is_mito)</code></pre>
<pre><code>## is_mito
## FALSE  TRUE 
## 32272    13</code></pre>
<pre class="r"><code>rowData(shamKidney)$symbol[is_mito]</code></pre>
<pre><code>##  [1] &quot;mt-Nd1&quot;  &quot;mt-Nd2&quot;  &quot;mt-Co1&quot;  &quot;mt-Co2&quot;  &quot;mt-Atp8&quot; &quot;mt-Atp6&quot; &quot;mt-Co3&quot; 
##  [8] &quot;mt-Nd3&quot;  &quot;mt-Nd4l&quot; &quot;mt-Nd4&quot;  &quot;mt-Nd5&quot;  &quot;mt-Nd6&quot;  &quot;mt-Cytb&quot;</code></pre>
<pre class="r"><code>shamKidney &lt;- addPerCellQC(shamKidney, subsets = list(mito=is_mito))
head(colData(shamKidney))</code></pre>
<pre><code>## DataFrame with 6 rows and 10 columns
##                    in_tissue array_row array_col   sample_id       sum
##                    &lt;logical&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;numeric&gt;
## AAACCGGGTAGGTACC-1      TRUE        42        28    sample01      6619
## AAACCGTTCGTCCAGG-1      TRUE        52        42    sample01      3675
## AAACCTCATGAAGTTG-1      TRUE        37        19    sample01     11004
## AAACGAGACGGTTGAT-1      TRUE        35        79    sample01     15085
## AAACTGCTGGCTCCAA-1      TRUE        45        67    sample01     16255
## AAAGGCTACGGACCAT-1      TRUE        62        54    sample01      3515
##                     detected subsets_mito_sum subsets_mito_detected
##                    &lt;integer&gt;        &lt;numeric&gt;             &lt;integer&gt;
## AAACCGGGTAGGTACC-1      2373              853                    11
## AAACCGTTCGTCCAGG-1      1616              457                    11
## AAACCTCATGAAGTTG-1      3317             1462                    11
## AAACGAGACGGTTGAT-1      4324             2153                    12
## AAACTGCTGGCTCCAA-1      3989             3147                    11
## AAAGGCTACGGACCAT-1      1611              523                    11
##                    subsets_mito_percent     total
##                               &lt;numeric&gt; &lt;numeric&gt;
## AAACCGGGTAGGTACC-1              12.8871      6619
## AAACCGTTCGTCCAGG-1              12.4354      3675
## AAACCTCATGAAGTTG-1              13.2861     11004
## AAACGAGACGGTTGAT-1              14.2725     15085
## AAACTGCTGGCTCCAA-1              19.3602     16255
## AAAGGCTACGGACCAT-1              14.8791      3515</code></pre>
<pre class="r"><code>plotVisium(shamKidney, fill = &quot;sum&quot;, trans = &quot;log&quot;, highlight = &quot;in_tissue&quot;)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Thresholds can be used as QC to remove spots that don’t meet metrics.
Including but not limited to:</p>
<ol style="list-style-type: decimal">
<li><p>Library size</p></li>
<li><p>Expressed genes</p></li>
<li><p>Proportion of gene specific reads (mitochondrial in this
example)</p></li>
<li><p>Cells per spot (If available for dataset)</p></li>
</ol>
<p>Appears to be a spike at low library sizes? An increased number of
spots with low number of transcripts around 1000 which can be filtered
out. Need to doublecheck the difference between UMIs and expressed
genes.</p>
<pre class="r"><code>hist(colData(shamKidney)$sum, breaks = 20) # Library Size, total sum of UMI counts per spot (included in sum column)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>hist(colData(shamKidney)$detected, breaks = 20) # Number of expressed genes per spot (detected column)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
<pre class="r"><code>hist(colData(shamKidney)$subsets_mito_percent, breaks = 20) # Proportion of mitochondrial genes per spot</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-5-3.png" width="672" /></p>
<pre class="r"><code># select QC threshold for library size
qc_lib_size &lt;- colData(shamKidney)$sum &lt; 1500
table(qc_lib_size)</code></pre>
<pre><code>## qc_lib_size
## FALSE  TRUE 
##  1731   169</code></pre>
<pre class="r"><code>colData(shamKidney)$qc_lib_size &lt;- qc_lib_size

# check spatial pattern of discarded spots
plotQC(shamKidney, type = &quot;spots&quot;, 
       discard = &quot;qc_lib_size&quot;)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-5-4.png" width="672" /></p>
<p>Remove and filter any low quality spots. Only based on UMIs</p>
<pre class="r"><code># combined set of discarded spots
discard &lt;- qc_lib_size
colData(shamKidney)$discard &lt;- discard

# filter low-quality spots
shamKidney &lt;- shamKidney[, !colData(shamKidney)$discard]</code></pre>
<p>Log based normalization can be performed to scale data approximately
and create clusters.</p>
</div>
<div id="clustering-and-visualization" class="section level1">
<h1>Clustering and Visualization</h1>
<pre class="r"><code>set.seed(123) #quick clustering for pool-based size factors
qclus &lt;- quickCluster(shamKidney)
# table(qclus)

shamKidney &lt;- computeSumFactors(shamKidney, cluster = qclus)
summary(sizeFactors(shamKidney))</code></pre>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1591  0.5492  0.9118  1.0000  1.3973  2.7630</code></pre>
<pre class="r"><code>hist(sizeFactors(shamKidney), breaks = 20)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>shamKidney &lt;-  logNormCounts(shamKidney)
assayNames(shamKidney)</code></pre>
<pre><code>## [1] &quot;counts&quot;    &quot;logcounts&quot;</code></pre>
<p>Feature selection based on highly variable genes (HVG) and spatially
variable genes (SVG). Should mitochondrial genes be removed? Are they
are likely highly expressed and/or are they of biological interest? SVG
selection based on SPARK-X package, HVG based on scran package</p>
<pre class="r"><code># shamKidney &lt;- shamKidney[!is_mito, ]
# dim(shamKidney)

deCells &lt;-  modelGeneVar(shamKidney)

fit &lt;-  metadata(deCells)

plot(fit$mean, fit$var, 
     xlab = &quot;mean of log-expression&quot;, ylab = &quot;variance of log-expression&quot;)
curve(fit$trend(x), col = &quot;dodgerblue&quot;, add = TRUE, lwd = 2)</code></pre>
<p><img src="SpatialVis_files/figure-html/getHVGs-1.png" width="672" /></p>
<pre class="r"><code>top_hvgs &lt;- getTopHVGs(deCells, prop = 0.1) #prop defines how many HVGS are wanted, 0.1 returns top 10% of genes
length(top_hvgs)</code></pre>
<pre><code>## [1] 446</code></pre>
<p>plot shows the total UMIs expressed in spots with differentially
expressed cells. This is the absolute numver of observed transcripts per
spot in spots expressing genes in the HVG list. (Based on 10x definition
of UMI, needs to be doublechecked)</p>
<p>Integration of highly variable genes and spatially variable genes can
be combines to further identify biologically relevant information not
cpatured by HVGs alone. Example in Li et al. 2021 combines principal
components of based on HVGs and SVGs in downstreams analysis.</p>
<pre class="r"><code>#Delete columns added in first addPerCellQC function to prevent duplicate columns
shamKidney$sum &lt;- shamKidney$detected &lt;- shamKidney$total &lt;-  NULL

#careful running multiple times, adds duplicate columns and breaks code.
shamKidney &lt;- addPerCellQC(shamKidney, subsets = list(hvg = top_hvgs))
# head(colData(shamKidney))

plotSpots(shamKidney, annotate = &quot;subsets_hvg_sum&quot;, palette = c(&quot;whitesmoke&quot;, &quot;red4&quot;))</code></pre>
<p><img src="SpatialVis_files/figure-html/visualizeHVGs-1.png" width="672" /></p>
<p>PCA and UMAP can be easily produced.</p>
<p>How do PCA/UMAP plots compare to the plots prpduced by loupe browser
for 6week sample? They are different maybe because the number of
clusters differ. Should they all be the same? How do they differ
compared to seurat clustering?</p>
<pre class="r"><code>set.seed(123)
shamKidney &lt;- runPCA(shamKidney, subset_row = top_hvgs)
shamKidney &lt;- runUMAP(shamKidney, dimred = &quot;PCA&quot;)

# dim(reducedDim(spe, &quot;PCA&quot;))

colnames(reducedDim(shamKidney, &quot;UMAP&quot;)) &lt;- paste0(&quot;UMAP&quot;, 1:2)

plotDimRed(shamKidney, type = &quot;PCA&quot;)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>plotDimRed(shamKidney, type = &quot;UMAP&quot;)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-8-2.png" width="672" /></p>
<p>Produces clusters of HVGs using graph-based clustering based on the
Walktrap method. Applies to top 50 PCs?</p>
<p>Change K values to change the number of clusters. Are less clusters
better or are more clusters better for analysis?</p>
<pre class="r"><code>set.seed(123)
k = 20
g &lt;- buildSNNGraph(shamKidney, k = k, use.dimred = &quot;PCA&quot;) #Graph based clustering
g_walk &lt;- igraph::cluster_walktrap(g)
clus &lt;- g_walk$membership
table(clus)</code></pre>
<pre><code>## clus
##   1   2   3   4   5   6   7   8 
## 208 216 727 306 136  45  36  57</code></pre>
<pre class="r"><code>colLabels(shamKidney) &lt;- factor(clus) #Store cluster labels in column &#39;label&#39; in colData

plotSpots(shamKidney, annotate = &quot;label&quot;,
          palette = rainbow(8))</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre class="r"><code>plotVisium(shamKidney, fill = &quot;label&quot;, trans = &quot;log&quot;, highlight = &quot;in_tissue&quot;, palette = rainbow(8))</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-9-2.png" width="672" /></p>
<pre class="r"><code>plotDimRed(shamKidney, type = &quot;PCA&quot;,
           annotate = &quot;label&quot;, palette = rainbow(8))</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-9-3.png" width="672" /></p>
<pre class="r"><code>plotDimRed(shamKidney, type = &quot;UMAP&quot;,
           annotate = &quot;label&quot;, palette = rainbow(8))</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-9-4.png" width="672" /></p>
<p>Code to identify potential marker genes, can also be done using
Seurat</p>
<pre class="r"><code>rownames(shamKidney) &lt;- rowData(shamKidney)$symbol

markers &lt;- findMarkers(shamKidney, test = &#39;binom&#39;, direction = &#39;up&#39;) #Produces list of clusters equal to clusters from graph based
markers</code></pre>
<pre><code>## List of length 8
## names(8): 1 2 3 4 5 6 7 8</code></pre>
<pre class="r"><code>interesting &lt;- markers[[4]] #selection for which cluster
best_set &lt;- interesting[interesting$Top &lt;= 5, ]
logFCs &lt;- getMarkerEffects(best_set)

pheatmap(logFCs, breaks = seq(-5, 5, length.out = 101))</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<pre class="r"><code>top_genes &lt;- head(rownames(interesting))
plotExpression(shamKidney, x = &quot;label&quot;, features = top_genes)</code></pre>
<p><img src="SpatialVis_files/figure-html/unnamed-chunk-10-2.png" width="672" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
